app/components/message-box.js

```diff
@@ -24,14 +24,18 @@ export default Ember.Component.extend({
   actions: {
     btn1Click() {
       if (this.btn1Callback) {
-        this.btn1Callback();
+        if (this.btn1Callback() === false) {
```

Comment:
Are there chances of having null or undefined value for `this.btn1Callback()`

Reply:
@Darshan-Chauhan the callbacks are initialized with empty methods, no risk

FYI this class is mostly a copy of the calendar for the public holidays selection. There are a few places using this date picker, but they're all slightly different. We should try to consolidate that into a single re-usable component